src/staking/manager.ts

```diff
@@ -0,0 +1,660 @@
+    inclusionProof: InclusionProof,
+  ): Promise<{
+    signedBbnTx: Uint8Array;
+  }> {
+    // Get the staking params at the time of the staking transaction
+    const p = getBbnParamByBtcHeight(stakingTxHeight, this.stakingParams);
+    if (!p)
+      throw new Error(
+        `Unable to find staking params for height ${stakingTxHeight}`,
+      );
+    
+    const stakerBtcInfo = await this.btcProvider.getStakerInfo();
+    if (!stakerBtcInfo) {
+      throw new Error("Staker info not found while registering delegation");
+    }
+
+    const bech32Address = await this.bbnProvider.getBbnAddress();
+    if (!bech32Address) {
+      throw new Error("Babylon address cannot be empty");
+    }
+
+    const stakingInstance = new Staking(
+      this.network,
+      stakerBtcInfo,
+      p,
+      stakingInput.finalityProviderPkNoCoordHex,
+      stakingInput.stakingTimelock,
+    );
+
+    // Create delegation message
+    const delegationMsg = await this.createBtcDelegationMsg(
+      stakingInstance,
+      stakingInput,
+      stakingTx,
+      bech32Address,
+      stakerBtcInfo,
+      p,
+      this.getInclusionProof(inclusionProof),
+    );
+
+    return {
+      signedBbnTx: await this.bbnProvider.signTransaction(delegationMsg),
+    };
+  }
+
+  /**
+   * Estimates the BTC fee required for staking.
+   * @param stakingInput - The staking inputs.
+   * @param feeRate - The fee rate in satoshis per byte.
+   * @returns The estimated BTC fee in satoshis.
+   */
+  async estimateStakingBtcFee(
+    stakingInput: StakingInputs,
+    feeRate: number,
+  ): Promise<number> {
+    const tipHeight = await this.bbnProvider.getBabylonBtcTipHeight();
+    if (tipHeight === 0) {
+      throw new Error("Babylon BTC tip height cannot be 0");
+    }
+    // Get the param based on the tip height
+    const p = getBbnParamByBtcHeight(tipHeight, this.stakingParams);
+    
+    const stakerInfo = await this.btcProvider.getStakerInfo();
+    if (!stakerInfo) {
+      throw new Error("Staker info not found while estimating staking fee");
+    }
+
+    const inputUTXOs = await this.btcProvider.getUTXOs();
+    if (inputUTXOs.length === 0) {
+      throw new Error("No input UTXOs provided");
+    }
+
+    const staking = new Staking(
+      this.network,
+      stakerInfo,
+      p,
+      stakingInput.finalityProviderPkNoCoordHex,
+      stakingInput.stakingTimelock,
+    );
+
+    const { fee: stakingFee } = staking.createStakingTransaction(
+      stakingInput.stakingAmountSat,
+      inputUTXOs,
+      feeRate,
+    );
+
+    return stakingFee;
+  }
+
+  /**
+   * Creates a signed staking transaction that is ready to be sent to the BTC 
+   * network.
+   * @param stakingInput - The staking inputs.
+   * @param unsignedStakingTx - The unsigned staking transaction.
+   * @param stakingParamVersion - The param version that was used to create the EOI
+   * @returns The signed staking transaction.
+   */
+  async createSignedBtcStakingTransaction(
```

Comment:
I wonder whether signing should be handled in this library or not. I'm not very comfortable with dependency injecting the btc provider and especially using it for signing operations. We are creating a weird link between the front-end wallet signing and the btc staking ts library that was not previously existent.

Ideally, this library would generate all transactions required and signing would be handled outside of it.